lib/tree.rb

```diff
@@ -1,38 +1,60 @@
 class NoApplesError < StandardError; end
 
-class AppleTree
-  attr_#fill_in :height, :age, :apples, :alive
+RANDOM = Random.new;
+
+class Tree
+  attr_reader :height, :age, :apples, :alive
 
   def initialize
+    @age = 0
+    @height = 1
+    @apples = []
+    @alive = true
   end
 
   def age!
+    @age += 1
+    if @alive
+      @height += 1
+      add_apples (@age * 2 - @apples.length)
+    end
+    if @age > 30
+      @alive = false
+    end
   end
 
-  def add_apples
+  def add_apples number
+    if @age >= 3
+      number.times { @apples.push(Apple.new( :red, RANDOM.rand(5)+3 )) }
```

Comment:
I tried that and had a problem. (I don't remember what it was). I tried it again, and it works fine. Probably  different error masked this output on my previous attempt.